sim/wait-for-it-quic/wait-for-it.go

```diff
@@ -32,39 +32,54 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
-	for range time.NewTicker(time.Second/2).C {
-		dur := time.Since(startTime)
-		if timeout > 0 && dur > timeout {
-			fmt.Printf("Failed to connect after %s\n", dur)
-			os.Exit(1)
-		}
-		if err := waitForIt(addr); err == nil {
-			break
-		}
-		time.Sleep(time.Second)
+	ok := waitForIt(addr, timeout)
+	dur := time.Since(startTime)
+	if !ok {
+		fmt.Printf("timeout occurred after waiting %s for %s\n", dur, host)
+		os.Exit(1)
 	}
-	fmt.Printf("%s is available after %s\n", host, time.Since(startTime))
+	fmt.Printf("%s is available after %s\n", host, dur)
 }
 
-func waitForIt(addr *net.UDPAddr) error {
-	conn, err := net.DialUDP("udp", nil, addr)
-	if err != nil {
-		return err
-	}
-	if _, err := conn.Write(packet); err != nil {
-		return err
-	}
-	b := make([]byte, 1500)
-	n, err := conn.Read(b)
+func waitForIt(addr *net.UDPAddr, timeout time.Duration) bool {
+	conn, err := net.ListenUDP("udp", nil)
 	if err != nil {
-		return err
+		log.Fatal("ListenUDP failed: %s", err)
 	}
-	b = b[:n]
-	if len(b) < 5 {
-		log.Fatal("invalid packet")
+	done := make(chan struct{})
+	go func() {
+		b := make([]byte, 1500)
+		n, err := conn.Read(b)
+		if err != nil {
+			log.Fatalf("Read failed: %s", err)
+			return
+		}
+		b = b[:n]
+		if len(b) < 5 {
+			log.Fatal("invalid packet")
+		}
+		if !bytes.Equal(b[1:5], []byte{0, 0, 0, 0}) {
+			log.Fatal("expected Version Negotiation packet")
+		}
+		close(done)
+	}()
+
+	ticker := time.NewTicker(time.Second / 2)
+	if timeout == 0 {
+		timeout = 365 * 24 * time.Hour
```

Comment:
What is the value of this timeout?

Reply:
Using a practically infinite value it no `-t` is given.

Reply:
You're right, it's ugly. Will fix.